packages/core/lib/util/get-response.ts

```diff
@@ -0,0 +1,139 @@
+import { ExecutionResult } from "graphql";
+import { ExecutionPatchResult } from "../types";
+import { calculateByteLength } from "./calculate-byte-length";
+import { ReadableStream, Response } from "cross-undici-fetch";
+
+export type TransformResultFn = (result: ExecutionResult | ExecutionPatchResult) => any;
+export const DEFAULT_TRANSFORM_RESULT_FN: TransformResultFn = (result: ExecutionResult) => result;
+
+export function getRegularResponse(executionResult: ExecutionResult, transformResult = DEFAULT_TRANSFORM_RESULT_FN): Response {
+  const transformedResult = transformResult(executionResult);
+  const responseBody = JSON.stringify(transformedResult);
+  const contentLength = calculateByteLength(responseBody);
+  const headersInit: HeadersInit = {
+    "Content-Type": 'application/json',
+    "Content-Length": contentLength.toString()
+  };
+  const responseInit: ResponseInit = {
+    headers: headersInit,
+    status: 200,
+  };
+  return new Response(responseBody, responseInit);
+}
+
+export function getMultipartResponse(
+  asyncExecutionResult: AsyncIterable<ExecutionPatchResult<any>>,
+  transformResult = DEFAULT_TRANSFORM_RESULT_FN
+): Response {
+  const headersInit: HeadersInit = {
+    Connection: "keep-alive",
+    "Content-Type": 'multipart/mixed; boundary="-"',
+    "Transfer-Encoding": "chunked",
+  };
+  const responseInit: ResponseInit = {
+    headers: headersInit,
+    status: 200,
+  };
+  const readableStream = new ReadableStream({
```

Comment:
ReadableStreams isnt support in workers yet, may i suggest using https://github.com/maraisr/piecemeal here for this.

Reply:
Could you share a source that shows that? I'd love to see it.

Reply:
Just try it :)